backend/lib/components/Oauth2.js

```diff
@@ -0,0 +1,35 @@
+const { Facebook } = require("./OAuth2/Facebook");
+const { Instagram } = require("./OAuth2/Instagram");
+const { Linkedin } = require("./OAuth2/Linkedin");
+const { Twitter } = require("./OAuth2/Twitter");
+const { Github } = require("./OAuth2/Github");
+const { SocialMock } = require("./OAuth2/SocialMock");
+const {
+  config: {
+    oauth2: { facebook, github, instagram, linkedin, twitter },
+  },
+} = require("../../config");
+
+// abstract factory to create oauth2 client
+const createOauth2Client = (socialType) => {
+  if (socialType === "facebook" && facebook.clientId && facebook.secretKey) {
```

Comment:
This could be more DRY if we defined an object mapping each `socialType` to the class:

```
const socialClasses = {
  facebook: {
    SocialClass: Facebook,
    config: facebook,
  },
  ...
};

...

const createOauth2Client = (socialType) => {
  if (!socialClasses.hasOwnProperty(socialType) {
    // Or maybe just return SocialMock since you defined that as a fallback.
    throw new Error('Unsupported socialType'); 
  }
  const { SocialClass, config } = socialClasses[socialType];
  return new SocialClass(config.clientId, config.secretKey);
}
```

Reply:
Thank you so much @mannykary for the comments! I will improve them 🚵 based on your advices